### PR TITLE
Extract tx regions

### PR DESCRIPTION
**Pull request by cmatKhan**
Adding the script for extracting the tx regions from the T2T gtf.

Note -- I added a line to the add_scripts.md with a brief oneliner about what the script I added does. Not sure if that was the intent of that file or not.

Related to:

```
Step (2)
Download a "gene regions" BED file with [tss, end] coordinates for each gene
```

### CODE REVIEW

**Comment by evanbiederstedt** [comment]
As I mentioned on slack:

In terms of organizing this, I think it's fine if people add code in whatever language for this (R, Python, etc. whatever)

Afterwards, we can polish things up. We'll make a Python package maybe, or R package---no reason we could do both and add C++

**Review by evanbiederstedt — CHANGES_REQUESTED** [review]
I just realized that we might have lots of code snippets in different languages

**Requests**

(A) If you have some time, could you make this an executable R script which uses argparse to read in an input and give an output?

That would be useful here in a few weeks. 

We then can have a `main()` function which runs everything in a way we expect. 

(B) Could you summarize all of the R package dependencies (versions too if you could) in the Markdown file? 


The above is motivated mostly by reproducibility/organization concerns so we don't run into issues in a few weeks

**Comment by evanbiederstedt** [comment]
Also, @cmatKhan please include the output BED files here as well, if you could

**Comment by cmatKhan** [comment]
[x] main method / cmd line optparsing
[x] data added to data/ -- data md updated
[x] add_scripts updated with notes on dependencies

File: scripts/create_tx_start_stop_bed.R [hunk 1]
```diff
@@ -0,0 +1,199 @@
+#'
```
Comment (evanbiederstedt) [inline scripts/create_tx_start_stop_bed.R]: `#!/usr/bin/env Rscript ` shebang here

File: scripts/create_tx_start_stop_bed.R [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+#' Parse a gtf file into objects for easy feature extraction
+#' https://www.ncbi.nlm.nih.gov/data-hub/genome/GCF_009914755.1/
```
Comment (evanbiederstedt) [inline scripts/create_tx_start_stop_bed.R:6]: Some variation of argparse would be useful here, e.g. 

https://blog.sellorm.com/2017/12/30/command-line-utilities-in-r-pt-6/
https://cran.r-project.org/web/packages/argparse/index.html

arguments: input, output

**Review by evanbiederstedt — APPROVED** [review]
HTML file looks nice: https://htmlpreview.github.io/?https://github.com/cmatKhan/isocomp/blob/extract_tx_regions/data/data.html

I just have other requests for the R script, i.e. shebang and explicitly using argparse

e.g. from this blog post: https://blog.sellorm.com/2017/12/30/command-line-utilities-in-r-pt-6/

```
library(argparser)

p <- arg_parser("The Sorting Hat")
# Add a positional argument
p <- add_argument(p, "name", help="Name of the person to sort")
# Add a flag
p <- add_argument(p, "--debug", help="enable debug mode", flag=TRUE)
# Add another flag
p <- add_argument(p, "--short", help="output only the house", flag=TRUE)
argv <- parse_args(p)
```

We'll need an input file and output file (probably use a default file, `output.bed`)

File: scripts/create_tx_start_stop_bed.R (hunk 2 above)
Comment (cmatKhan) [reply scripts/create_tx_start_stop_bed.R:6]: I'm using optparse -- are you saying you'd prefer argparse?

Comment (cmatKhan) [reply scripts/create_tx_start_stop_bed.R:6]: This is what running looks like:

```
➜  isocomp git:(extract_tx_regions) scripts/create_tx_start_stop_bed.R --help
Usage: scripts/create_tx_start_stop_bed.R [options]


Options:
	-g GTF, --gtf=GTF
		path to the gtf annotation set. Suggested source:  https://www.ncbi.nlm.nih.gov/data-hub/genome/GCF_009914755.1/

	-t TXDB, --txdb=TXDB
		path to a bioconductor sqlite TxDb. If one DNE, it will be created. In that case, provide a writeable path.

	-h, --help
		Show this help message and exit
```

I would suggest not adding the GTF input since its available from NCBI through that link -- thoughts?

A run command looks like this:

```
scripts/create_tx_start_stop_bed.R -g data/GCF_009914755.1/genomic.sorted.gtf -t data/t2t_txdb.sqlite
```
I suppose I should put that into the add_scripts.md
